test/with_dummyserver/test_https.py

```diff
@@ -439,9 +438,9 @@ def test_assert_invalid_fingerprint(self):
                                         'AA:AA:AA:AA:AA:AA:AA:AA:AA'
 
         def _test_request(pool):
-            with self.assertRaises(MaxRetryError) as cm:
+            with pytest.raises(MaxRetryError) as cm:
                 pool.request('GET', '/', retries=0)
-            self.assertIsInstance(cm.exception.reason, SSLError)
+                assert isinstance(cm.exception.reason, SSLError)
```

Comment:
This assert can never run, because the `pool.request()` call raises. You need to remove those two spaces to execute this assert *after* the with block. And yes, `cm` will still be defined at this point. I agree that it's confusing! But [this is how it's intended to work](https://docs.pytest.org/en/latest/assert.html#assertraises).